ipi/utils/parsing.py

```diff
@@ -219,13 +219,37 @@ def read_trajectory(
                 if len(matches) >= 1:
                     frame.info["step"] = int(matches[-1][0])
 
-                # if we have forces, set positions to zero (that data is missing!) and set forces instead
-                if what == "forces":
-                    # set forces and convert to eV/angstrom
-                    frame.positions *= 0
-                    frame.arrays["forces"] = ret["atoms"].q.reshape(
-                        (-1, 3)
-                    ) * unit_to_user("force", "ev/ang", 1.0)
+                # if we have forces, set positions to zero (that data is missing!)
+                # and set forces instead
+                if what in [
+                    "positions",
+                    "x_centroid",
+                    "x_centroid_even",
+                    "x_centroid_odd",
+                ]:
+                    pass  # nothing to do here, positions is the right place to store
+                else:
+                    frame.positions *= 0.0  # trajectory does NOT contain position data!
+                    frame.arrays[what] = ret["atoms"].q.reshape((-1, 3))
+                    if what in [
+                        "forces",
+                        "forces_spring",
+                        "forces_component",
+                        "forces_sc",
+                        "Eforces",
+                        "f_centroid",
+                        "forces_component_raw",
```

Comment:
so every time we add a new position keyword of force keyword we have to hand-modify this list, yes? maybe worth adding a disclaimer about this in the file.

Reply:
Yes, I don't see an easy way around it unfortunately.

Reply:
Actually I didn't think hard enough. Thanks for spotting this, there as indeed a way to avoid the manual mess.